examples/quilkin-filter-example/src/main.rs

```diff
@@ -55,22 +63,16 @@ impl FilterFactory for GreetFilterFactory {
         NAME
     }
     fn create_filter(&self, args: CreateFilterArgs) -> Result<Box<dyn Filter>, Error> {
-        let greeting = match args.config.unwrap() {
-            ConfigType::Static(config) => {
-                serde_yaml::from_str::<Config>(serde_yaml::to_string(config).unwrap().as_str())
-                    .unwrap()
-                    .greeting
-            }
-            ConfigType::Dynamic(config) => {
-                let config: greet::Greet = prost::Message::decode(Bytes::from(config.value)).unwrap();
-                config.greeting
-            }
-        };
-        Ok(Box::new(Greet(greeting)))
+        let greeting = self
+            .require_config(args.config)?
+            .deserialize::<Config, ProtoGreet>(self.name())?;
+        Ok(Box::new(Greet(greeting.greeting)))
```

Comment:
this is now out of sync with the doc, we'll need to update that to match

Reply:
That'll be my next job! I'll take a stab at pulling the code samples from this code directly as well via mdbook. 👍🏻